app/scripts/controllers/deploymentConfig.js

```diff
@@ -87,6 +98,34 @@ angular.module('openshiftConsole')
             $scope.deploymentConfig = deploymentConfig;
             updateHPAWarnings();
             ImageStreamResolver.fetchReferencedImageStreamImages([deploymentConfig.spec.template], $scope.imagesByDockerReference, $scope.imageStreamImageRefByDockerReference, context);
+            copyDeploymentConfigAndEnsureEnv(deploymentConfig);
+            $scope.saveEnvVars = function() {
+              // copy again to clean off any of angulars $$ vars from the template
+              var updatedDeploymentConfig = angular.copy($scope.updatedDeploymentConfig);
+              updatedDeploymentConfig.spec.template.spec.containers = _.map(updatedDeploymentConfig.spec.template.spec.containers, function(container) {
+                container.env = _.reject(container.env, function(env){
+                  return env.name === "" && env.value ==="";
```

Comment:
actually just `!env.name` this never got updated to only reject when the name is empty, you may want to set an env to an empty value